pkg/bundle/sync.go

```diff
@@ -138,41 +139,103 @@ func (b *bundle) buildSourceBundle(ctx context.Context, bundle *trustapi.Bundle)
 
 // configMapBundle returns the data in the source ConfigMap within the trust Namespace.
 func (b *bundle) configMapBundle(ctx context.Context, ref *trustapi.SourceObjectKeySelector) (string, error) {
-	var configMap corev1.ConfigMap
-	err := b.client.Get(ctx, client.ObjectKey{Namespace: b.Namespace, Name: ref.Name}, &configMap)
-	if apierrors.IsNotFound(err) {
-		return "", notFoundError{err}
-	}
+	if ref.Name != "" {
+		var configMap corev1.ConfigMap
+		err := b.client.Get(ctx, client.ObjectKey{Namespace: b.Namespace, Name: ref.Name}, &configMap)
 
-	if err != nil {
-		return "", fmt.Errorf("failed to get ConfigMap %s/%s: %w", b.Namespace, ref.Name, err)
-	}
+		if apierrors.IsNotFound(err) {
+			return "", notFoundError{err}
+		}
 
-	data, ok := configMap.Data[ref.Key]
-	if !ok {
-		return "", notFoundError{fmt.Errorf("no data found in ConfigMap %s/%s at key %q", b.Namespace, ref.Name, ref.Key)}
-	}
+		if err != nil {
+			return "", fmt.Errorf("failed to get ConfigMap %s/%s: %w", b.Namespace, ref.Name, err)
+		}
+
+		data, ok := configMap.Data[ref.Key]
+		if !ok {
+			return "", notFoundError{fmt.Errorf("no data found in ConfigMap %s/%s at key %q", b.Namespace, ref.Name, ref.Key)}
+		}
+		return data, nil
+	} else {
+		var configMapList corev1.ConfigMapList
+
+		labelSelectorRequirements, err := labels.ParseToRequirements(metav1.FormatLabelSelector(ref.Selector))
```

Comment:
I think we should rather use `metav1.LabelSelectorAsSelector` here - to get better/safer error handling. The function `metav1.FormatLabelSelector` seems to return "funny" strings in case of error. Maybe create a small helper to reduce duplication even more?

Reply:
I refactored this part a bit to reduce duplication and used `metav1.LabelSelectorAsSelector`. Let me know what you think!